### PR TITLE
Update the telemetry for msaz

### PR DESCRIPTION
**Pull request by NoriZC**
# PR Summary

- Updated Installation Id Initialization to support both AzPS and AzCLI on Unbuntu. Tested on both ubuntu dockers for azps and azcli.
  - AzCLI:
  - image
  
  - AzPSH:
  - image

### CODE REVIEW

**Comment by daxian-dbw** [comment]
@NoriZC Please revert the changes in `AzureAgent.cs`. We will put off telemetry code changes until the metrics for each telemetry are well defined.

**Comment by NoriZC** [comment]
> @NoriZC Please revert the changes in `AzureAgent.cs`. We will put off telemetry code changes until the metrics for each telemetry are well defined.

OK, I will seperate this PR soon

File: shell/agents/Microsoft.Azure.Agent/Telemetry.cs [hunk 1]
```diff
@@ -17,10 +17,10 @@ internal static void Initialize()
...
         InstallationId = null;
 
-        string azureConfigDir = Environment.GetEnvironmentVariable("AZURE_CONFIG_DIR")
-            ?? Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), ".azure");
-        string azCLIProfilePath = Path.Combine(azureConfigDir, "azureProfile.json");
-        string azPSHProfilePath = Path.Combine(azureConfigDir, "AzureRmContextSettings.json");
+        string azureConfigDir = Environment.GetEnvironmentVariable("AZURE_CONFIG_DIR");
+        string userProfile = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
+        string azCLIProfilePath = azureConfigDir ?? Path.Combine(userProfile, ".azure", "azureProfile.json");
+        string azPSHProfilePath = azureConfigDir ?? Path.Combine(userProfile, ".Azure", "AzureRmContextSettings.json");
```
Comment (daxian-dbw) [inline shell/agents/Microsoft.Azure.Agent/Telemetry.cs:23]: `azureConfigDir` is a folder path, so when `AZURE_CONFIG_DIR` is defined in environment variable, `azCLIProfilePath` and `azPSHProfilePath` will be set to the folder path in your code.

**Review by daxian-dbw — CHANGES_REQUESTED** [review]

**Comment by daxian-dbw** [comment]
> OK, I will seperate this PR soon

Do not work on the new changes yet. We need to metrics to be well defined before start coding work. We don't want to make changes again and again.

**Comment by daxian-dbw** [comment]
@NoriZC You can continue with fixing installation id in this PR, but please revert the unrelated changes.

**Review by daxian-dbw — APPROVED** [review]

**Pull request merged**